File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/CurriculumSectionDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * DTO for a curriculum section.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CurriculumSectionDTO {
    private String title;
    private Integer index;
    private String duration; // e.g., "15:47"
    private Integer durationSeconds;
    private Integer lectureCount;
    private List<CurriculumItemDTO> items;
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/CurriculumSectionDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for a curriculum section.
 */
data class CurriculumSectionDTO(
    val title: String? = null,
    val index: Int? = null,
    // e.g., "15:47"
    val duration: String? = null,
    val durationSeconds: Int? = null,
    val lectureCount: Int? = null,
    val items: List<CurriculumItemDTO>? = null,
)
